### PR TITLE
Update After Effects and Photoshop plug-ins

### PR DESCRIPTION
**Pull request by fnordware**
Keeping up with OCIO library changes.

### CODE REVIEW

File: vendor/photoshop/OpenColorIO_PS_Context.cpp
Comment (doug-walker) [inline vendor/photoshop/OpenColorIO_PS_Context.cpp:155]: Recommend the default (VERY_GOOD).

**Review by doug-walker — APPROVED** [review]
Hi Brendan, thanks for updating these!

I'm not the ideal person to review this, but these are your plug-ins and I want to unblock you so you could merge your changes.  I made a few minor comments under the assumption that you are trying to just get the plug-ins working with the new library but not necessarily take advantage of all the new v2 features.

I'm curious if you are planning to do a more extensive follow-up PR that would take advantage of the new v2 features?  For example, I noticed that you have some ICC Profile related code and it might be interesting to use the new ICC features in v2.  Also, v2 now supports integer buffers directly, which might be a useful performance boost for you.  It would also be nice to support the category feature to simplify menus for artists.  Glad to see you're using the new GPU renderer though!  Patrick and I could have a Zoom mtg with you if you'd like to discuss the other items.

**Review by hodoulp — APPROVED** [review]
Thanks @fnordware to follow the OCIOv2 development.

File: vendor/aftereffects/OpenColorIO_AE_Context.cpp
Comment (fnordware) [reply vendor/aftereffects/OpenColorIO_AE_Context.cpp:659]: Hey Doug, with regards to this specific comment, I do use the default optimization in general, but this is a little feature I added where the user can use LOSSLESS (i.e. EXACT) if they want to. It's just an option, and a fairly well-hidden one at that.

But I would be happy to have a chat with you about other OCIO v2 features. I did experiment a little with categories but I don't fully understand them and don't have any v2 configurations to test with. Maybe you could educate me. I tried the integer buffers at one point and found they were actually slower. Maybe I should try them again.

File: vendor/aftereffects/mac/OpenColorIO_AE_MonitorProfileChooser_Controller.m
Comment (fnordware) [reply vendor/aftereffects/mac/OpenColorIO_AE_MonitorProfileChooser_Controller.m:6]: Apparently not…

File: vendor/photoshop/OpenColorIO_PS_Context.cpp
Comment (fnordware) [reply vendor/photoshop/OpenColorIO_PS_Context.cpp:155]: I figured for Photoshop I may as well stick with LOSSLESS because the user is only rendering one frame, but I'm happy to take your recommendation.

File: vendor/aftereffects/OpenColorIO_AE_Context.cpp
Comment (doug-walker) [reply vendor/aftereffects/OpenColorIO_AE_Context.cpp:659]: It's totally fine to allow different optimization levels.  Sorry, I didn't realize you had done that.

Given that Photoshop is often starting with integer pixel buffers, you should be getting a big performance boost for many color space conversions.  For example, in the ACES config, doing a ColorSpaceTransform where the source space is one of the log color spaces that starts with an inverse Lut1D, it should be a lot faster.  But you would need to be VERY_GOOD rather than LOSSLESS to get the speed-up.  There is also a stand-alone tool called ocioperf that may be useful when looking at performance issues.

There is a v2 demo config in the documentation now, so you could take a look at that.  We'd be happy to chat further at any time, DM me on the OCIO Slack when you're ready and we could set up a time.

**Comment by doug-walker** [comment]
@fnordware, we were not sure if you wanted to do more in this branch or merge this one and continue in a follow-up.  Whenever you'd like us to merge this, just update the branch with master and ping us and we will merge it.

File: vendor/aftereffects/mac/OpenColorIO_AE_MonitorProfileChooser_Controller.m
Comment (fnordware) [reply vendor/aftereffects/mac/OpenColorIO_AE_MonitorProfileChooser_Controller.m]: Removed.

**Review by michdolan — APPROVED** [review]

**Comment by hodoulp** [comment]
Do not merge. We are waiting for the 'go' from @fnordware.

**Comment by fnordware** [comment]
OK, guys, I think I've done all I intend to do to these plug-ins for now. Merge away.

**Comment by hodoulp** [comment]
@fnordware Great. I will merge it.

**Pull request merged**
